pkg/storage/replica.go

```diff
@@ -1936,10 +1951,10 @@ func (r *Replica) tryAddWriteCmd(
 			if tryAbandon() {
 				log.Warningf(ctx, "context cancellation after %0.1fs of attempting command %s",
 					timeutil.Since(startTime).Seconds(), ba)
-				// Set endCmds to nil because they will be invoked in processRaftCommand.
-				endCmds = nil
 				return nil, roachpb.NewError(roachpb.NewAmbiguousResultError(ctx.Err().Error())), proposalNoRetry
 			}
+			// Set endCmds to nil because they will be invoked in processRaftCommand.
```

Comment:
Well, the previous code was correct in one area but broken in another. And this change is correct in one area but broken in another. I've got my head wrapped around the problem. Let me go back to the drawing board.